branchpro/apps/_reproduction_number_plot.py

```diff
@@ -0,0 +1,139 @@
+#
+# ReproductionNumberPlot Class
+#
+# This file is part of BRANCHPRO
+# (https://github.com/SABS-R3-Epidemiology/branchpro.git) which is released
+# under the BSD 3-clause license. See accompanying LICENSE.md for copyright
+# notice and full license details.
+#
+
+import warnings
+import pandas as pd
+import plotly.graph_objs as go
+
+
+class ReproductionNumberPlot():
+    """ReproductionNumberPlot Class
+    Stores the main figure for the Dash app.
+    """
+    def __init__(self):
+        self.figure = go.Figure()
+
+    def _label_warning(self, time_key, r_key):
+        x_label = self.figure['layout']['xaxis']['title']['text']
+        y_label = self.figure['layout']['yaxis']['title']['text']
+        if (x_label is not None) and (y_label is not None):
+            if (x_label != time_key) or (y_label != r_key):
+                warnings.warn('Labels do not match. They will be updated.')
+
+    def add_ground_truth_rt(self, df, time_key='Time Points', r_key='R_t'):
+        """
+        Plots the true values of R_t as a line on the figure.
+
+        Parameters
+        ----------
+        df
+            (pandas DataFrame) contains the true values of the reproduction
+            number by days. Data stored in columns 'Time Points' and 'R_t',
+            respectively.
+        time_key
+            x-axis label for the bar plot.
+        r_key
+            y-axis label for the bar plot.
+        """
+        if not issubclass(type(df), pd.DataFrame):
+            raise TypeError('df needs to be a dataframe')
+        self._label_warning(time_key, r_key)
+
+        trace = go.Scatter(
+            y=df[r_key],
+            x=df[time_key],
+            mode='lines',
+            name='True R',
+            line_color='green'
+        )
+
+        self.figure.add_trace(trace)
+        self.figure.update_layout(
+            xaxis_title=time_key,
+            yaxis_title=r_key)
+
+    def add_interval_rt(
+            self, df, time_key='Time Points', r_key='Mean',
+            lr_key='Lower bound CI', ur_key='Upper bound CI'):
```

Comment:
It would be nice if this method displayed the probability of the interval in the legend.

As it stands, that information is not saved in the dataframe which is returned by inference, though that could be changed.

Alternatively, we could expect the user to supply that information as an optional argument to this method.